src/server/game/Entities/AreaTrigger/AreaTrigger.cpp

```diff
@@ -642,6 +659,60 @@ bool AreaTrigger::HasSplines() const
     return bool(_spline);
 }
 
+void AreaTrigger::InitCircularMovement(AreaTriggerCircularMovementInfo const& cmi, uint32 timeToTarget)
+{
+    // should be sent in object create packets only
+    m_uint32Values[AREATRIGGER_TIME_TO_TARGET] = timeToTarget;
+
+    _areaTriggerCircularMovementInfo = cmi;
+
+    _areaTriggerCircularMovementInfo.TimeToTarget = GetTimeToTarget();
+    _areaTriggerCircularMovementInfo.ElapsedTimeForMovement = GetElapsedTimeForMovement();
+
+    if (IsInWorld())
+    {
+        // TODO: broadcast AreaTriggerReShape?
+    }
+}
+
+Position const* AreaTrigger::GetCircularMovementCenterPosition() const
+{
+    if (!_areaTriggerCircularMovementInfo.Center.is_initialized() && _areaTriggerCircularMovementInfo.TargetGUID.is_initialized())
+        if (WorldObject* center = ObjectAccessor::GetWorldObject(*this, *_areaTriggerCircularMovementInfo.TargetGUID))
+            return center;
+
+    return _areaTriggerCircularMovementInfo.Center.is_initialized() ? &_areaTriggerCircularMovementInfo.Center->Pos : nullptr;
+}
+
+void AreaTrigger::UpdateCircularMovementPosition()
+{
+    Position const* centerPos = GetCircularMovementCenterPosition();
+    if (!centerPos)
+        return;
+
+    AreaTriggerCircularMovementInfo const& cmi = _areaTriggerCircularMovementInfo;
+
+    // AreaTrigger make exactly "TimeToTarget / Duration" loops during his life time
```

Comment:
`TimeToTarget > Duration` is that ever true?

Reply:
Can look later if that's ever true

Reply:
Yes its true.. look at the sql traesh provided.. there is a example.. this spell does have a duration of 21 sec but does have a timeToTarget of 31 sec

Reply:
CircleCount/TimeToTarget = TimePerCircle

Reply:
wtf is CircleCount?

Reply:
The Amount of Circles you want :D just for clarification.. just put a 1 in there.. TimeToTarget / Duration = CircleCount

Reply:
then the example does about 1,5 circles during its lifetime?

Reply:
Exactly, these areatriggers are spawned at ymiron in maw of souls.. he spawns them with different radius variables, which results in bigger/smaller circles, which results in bigger/smaller TimeToTarget values

Reply:
then the implementation is wrong, it's Duration/TimeToTarget, what it right now?

Reply:
the implementation is Duration/TimeToTarget, the comment says TimeToTarget/Duration

Reply:
You mean bigger/smaller Circles resulting in bigger/smaller TimeToTarget Values? thats only the case for Ymiron.. he does that in his script.. that has nothing do to with the system itself

Reply:
Yeah the comment is wrong, i will change it